core/src/impl/Kokkos_GraphImpl.hpp

```diff
@@ -40,6 +40,17 @@ struct GraphAccess {
         std::make_shared<GraphImpl<ExecutionSpace>>(std::move(ex))};
     //----------------------------------------//
   }
+
+#if defined(KOKKOS_ENABLE_CUDA)
+  template <class ExecutionSpace, typename T>
+  static Kokkos::Experimental::Graph<ExecutionSpace>
+  construct_graph_from_native(ExecutionSpace ex, T&& native_graph) {
```

Comment:
same as above.

Reply:
This is in `Impl::` isn't it?
Unless I am mistaken then I would rather not have the hard to read constraint on here.

Reply:
This is even more hidden than what you think :wink:
```c++
Kokkos::Impl::GraphAccess::construct_graph_from_native
```

So yes, I would not bother too much with the constraints here.